packages/lib/src/components/internal/Countdown/index.tsx

```diff
@@ -8,75 +12,55 @@ interface CountdownProps {
 }
 
 interface CountdownState {
-    startTime: Date;
-    endTime: Date;
     minutes: string;
     seconds: string;
 }
 
-class Countdown extends Component<CountdownProps, CountdownState> {
-    constructor(props) {
-        super(props);
-        const secondsFromNow = this.props.minutesFromNow * 60000;
-        const nowTime = new Date().getTime();
-        this.state = {
-            startTime: new Date(nowTime),
-            endTime: new Date(nowTime + secondsFromNow),
-            minutes: '-',
-            seconds: '-'
-        };
-    }
-
-    public static defaultProps = {
-        onTick: () => {},
-        onCompleted: () => {}
-    };
-
-    protected interval;
+function Countdown({ minutesFromNow, onTick = () => {}, onCompleted = () => {} }: CountdownProps) {
+    const secondsFromNow = minutesFromNow * 60000;
+    const nowTime = new Date().getTime();
+    const startTime = new Date(nowTime);
+    const endTime = new Date(nowTime + secondsFromNow);
 
-    tick() {
-        const newTime = getTimeDifference(this.state.startTime, this.state.endTime);
+    const { i18n } = useCoreContext();
+    const { srPanel } = useSRPanelContext();
 
-        if (newTime.completed) {
-            this.props.onCompleted();
-            return this.clearInterval();
-        }
+    const a11yManager = useRef(null);
```

Comment:
Minor thing - just sharing: the other idea that we discussed before, more "reactish" way, would be to use custom hook for this:

```js
const { updateCoutdownSrPanel,  teardownCoutdownSrPanel } = useCoutdownA11yReporter({ srPanel, i18n });
```

The syntax could be less verbose IMO. Instead of `a11yManager.current.update(time)` you could do `updateCoutdownSrPanel(time)`

Reply:
This style is indeed less verbose and easy to read, but I am struggling to find a proper way to use/implement it.
I need to provide the context (initiate the context) somewhere. As a result, the mark down will become more verbose.. Or there is another way to do so?
```html
<CoutdownA11yContext.Provider value={{ srPanel, i18n }}>
       <Countdown />
</CoutdownA11yContext.Provider>
```
However, I could implement something like this:
```javascript
function useCoutdownA11yReporter({ srPanel, i18n }) {
    const a11y = new A11yManager({ srPanel, i18n });
    return { updateCoutdownSrPanel: a11y.update, teardownCoutdownSrPanel: a11y.tearDown };
}
```
Furthermore `useCoutdownA11yReporter` will be called only once on mount.

Reply:
Why would you need a context for it? I believe you can do it with custom hook without providing context/provider.

Context/Provider is useful when you have to pass things down the tree for different Components, avoiding prop drilling. In this case, seems like we do not need it,

Reply:
I read it wrongly somehow, hence I was thinking about the provider/context.